### ISSUE TITLE
x/telemetry/upload: log upload related activities for debug and inspection

### ISSUE THREAD

**Issue by hyangah** [golang/go#63709]
Upload logic involves some complex operations and actual upload occurs once a week.
Leave logging on disk will help users inspect and debug.

* Should this be always on, or opt-in?
* Should logs rotate and gc'd periodically?

**Comment by pjweinb** [comment]
This needs to be decided by the telemetry team.

**Comment by pjweinb** [comment]
This needs to be decided by the team. No point in assigning it until that happens.

**Comment by findleyr** [comment]
We discussed this briefly in the telemetry meeting, but our consensus was under-specified, unfortunately.

We decided that if the `<config>/go/telemetry/debug` folder were present, the telemetry upload would log to that location. However, in the discussion of https://go.dev/cl/558437, it became apparent that there wasn't agreement whether this logic belongs in x/telemetry, or at the call site (e.g. gopls). In other words: should the caller be checking for the `debug` folder, by convention, or should `x/telemetry` implement this debugging logic automatically.

In my opinion, it is better for this logic to be implemented in [`x/telemetry.Upload`](https://pkg.go.dev/golang.org/x/telemetry/upload), because then:

1. We don't need to reimplement it in each caller.
2. For the purposes of understanding uploading on my local machine, I can just create the `debug` folder and be guaranteed to observe all the uploads that occur, from any process.
3. If there is ever a garbage collection process, we can implement it in x/telemetry. If control is delegated to the caller, the caller would need to run its own garbage collection.
4. We don't have to expose the location of telemetry folders in the public API, as was necessary in https://go.dev/cl/558437.  By moving the debug logic into `x/telemetry`, we don't require any new public APIs, and are therefore free to change this behavior in the future.

To be precise, this is how I imagine it working:

- Callers can continue to provide [`upload.Control.Logging`](https://pkg.go.dev/golang.org/x/telemetry/upload#Control.Logging), and if set the upload will log to that output. No change there.
- If the `os.UserConfigDir()/go/telemetry/debug` folder is present, callers will *also* write a log to that directory. Here I don't feel strongly about how those logs are structured: there could be one log, one log per day, one log per process, etc. Whatever works best.
- We don't need to worry about rotation or GC, at least not initially, since we can just delete the folder after completing our investigation.

CC @hyangah @pjweinb @adonovan, to hopefully reach a consensus. Any objections to this approach?
